### ISSUE TITLE
RangeError: Maximum call stack size exceeded when seeding Prisma database

### ISSUE THREAD

**Issue by GitMarijn** [prisma/prisma#13938]
### Bug description

When running the  `"db:seed": "NODE_ENV=test NODE_PATH='.' dotenv -e .env.local -- ts-node-dev db/seeds/seed.ts",` I keep getting a `RangeError: Maximum call stack size exceeded` error. 

Tried downgrading TypeScript, node and ts-node. Unfortunately this didn't help.

### How to reproduce

1. Run the `"prisma": {
    "schema": "db/schema.prisma",
    "seed": "npm run db:seed"
  },` command
2. Get error: ```/node_modules/typescript/lib/typescript.js:68629
                        invokeOnce(source, target, inferFromObjectTypes);
                        ^
RangeError: Maximum call stack size exceeded
    at inferFromTypes (/node_modules/typescript/lib/typescript.js:68629:25)
    at inferFromTypes (/node_modules/typescript/lib/typescript.js:68449:25)
    at inferFromTypes (/node_modules/typescript/lib/typescript.js:68449:25)
    at inferFromProperties (/node_modules/typescript/lib/typescript.js:68992:25)
    at inferFromObjectTypes (/node_modules/typescript/lib/typescript.js:68980:21)
    at invokeOnce (/node_modules/typescript/lib/typescript.js:68661:21)
    at inferFromTypes (node_modules/typescript/lib/typescript.js:68629:25)
    at inferFromTypes (/node_modules/typescript/lib/typescript.js:68449:25)
    at inferFromProperties (/node_modules/typescript/lib/typescript.js:68992:25)
    at inferFromObjectTypes (/node_modules/typescript/lib/typescript.js:68980:21)```

### Expected behavior

Running the db:seed command without errors.

### Prisma information

Error seems to occur in this model. When removing relations with `PackageStep`, `PackageTranslation`, `editedBy` and `createdBy` the error was gone. But this is obviously not a feasible solution.

```model Package {
  id   Int     @id @default(autoincrement())
  uuid String  @unique @default(uuid())
  data String? @db.LongText

  status   Status @relation(fields: [statusId], references: [id])
  statusId Int    @map("mrb_sys_status_id")

  companyProduct   CompanyProduct @relation(fields: [companyProductId], references: [id])
  companyProductId Int            @map("mrb_sys_company_product_id")

  createdAt DateTime @default(now()) @map("created")
  editedAt  DateTime @updatedAt @map("edited")

  createdById Int  @map("created_by_id")
  createdBy   User @relation(name: "createdPackages", fields: [createdById], references: [id])

  editedById Int  @map("edited_by_id")
  editedBy   User @relation(name: "editedPackages", fields: [editedById], references: [id])

  cartItems          CartItem[]           @relation("package")
  campaigns          CampaignPackage[]
  calendarEventLinks CalendarEventLink[]
  media              Media[]
  translations       PackageTranslation[]
  steps              PackageStep[]

  @@map("mrb_module_package")
}
```

### Environment & setup

- OS: macOS 12.4
- Database: MySQL 5.6.51
- Node.js version: 16.13.0

### Prisma Version

```
3.14.0
```

**Comment by wildky** [comment]
@GitMarijn @matthewmueller Possibly relevant: I was getting this error too when trying to seed about ~300,000 rows. I was confused because I was able to do it just a few days before, the only difference this time was that I added a couple of new properties to the prisma schema (and to the data I was trying to insert via the seed).

I may be wrong, but after a bit of troubleshooting I believe I was getting this error because I had many type errors in the data I was trying to seed with the `createMany` command, and prisma was trying to report all of these type errors to me in an error message, but it ran into a limit while trying to list all of them in the console (i.e. the `RangeError`). 

What was unfortunate about this situation was that the helpful type errors were hidden to me until the entire process ran (i.e. Prisma prints out all of the type errors only after it tries to insert all of the data) -- so when I was inserting _a lot_ of data, I only reached the unhelpful `RangeError` message before I saw the actually relevant errors about my data not matching the Prisma schema. But seeing the type errors would have let me fix the problem immediately. 

While troubleshooting, I reduced the amount of data I was inserting (~50,000 rows vs ~250,000 rows) and after waiting ~5 minutes (usually the seed runs in ~10 seconds) I got a massive list of errors about mis-matching types in data I was trying to insert. 

Once I fixed the data I was trying to insert during the seed, the original seed ran successfully (and in just a few seconds!)

So it seems like Prisma trying to report all of the type errors (1) was a very time intensive task -- orders of magnitude more than actually running the large seed, and (2) blocked me from actually seeing the the helpful error messages because it reached the RangeError while trying to report the _actual_ errors. 

I found a few github issues of people reporting seeing a similar `RangeError: Maximum call stack size exceeded` when running a seed, but I didn't really see any definitive resolutions. I wonder if this issue I ran into is the same issue that others are running into, or if there may be other reasons this `RangeError` can come up during a seed. Either way, this could have been avoided for me if I somehow knew there were type errors in the data I was trying to insert before the seed tried to insert all of the data. For example, would it be possible Prisma only reported the first ~10 errors if there is a massive list of them, instead of trying to print all 10's or 100's of thousands of messages?

Not sure if this makes sense, but this was my interpretation of the situation and thought I'd chime in.

**Comment by SevInf** [comment]
I was not able to reproduce this issue. See the attempt here: https://github.com/prisma/reproductions/tree/main/gh-13938

For me, it works without major issues for 1000 errors. At 10000 errors typechecking takes forever, but eventually finishes the exception. @wildky it would really help us if you could share your schema and seeding code so we could have another attempt at reproducing it.

Regarding other `RangeError` reports, there are 2 kind of them at the moment. The one which is reported here, causes TS compiler crash. Other one, #9372 causes a crash in prisma runtime itself. That one I was able to reproduce in #15153.

**Comment by wildky** [comment]
> I was not able to reproduce this issue. See the attempt here: https://github.com/prisma/reproductions/tree/main/gh-13938
> 
> For me, it works without major issues for 1000 errors. At 10000 errors typechecking takes forever, but eventually finishes without the exception. @wildky it would really help us if you could share your schema and seeding code so we could have another attempt at reproducing it.
> 
> Regarding other `RangeError` reports, there are 2 kind of them at the moment. The one which is reported here, causes TS compiler crash. Other one, #9372 causes a crash in prisma runtime itself. That one I was able to reproduce in #15153.

@SevInf It didn't crash for me either even when trying to insert 50,000 rows with errors. It did crash for 250,000 rows. 

Since you only tried up to 10,000 rows, I don't think you tried inserting enough data to get it to crash. 

I'm happy to create and share some code that reproduces the error if necessary, but could you try inserting more erroneous data in your environment that you already have first? It sounds like you just aren't inserting enough to get the crash yet (250,000 rows is the best guess I have to get it to crash)

**Comment by SevInf** [comment]
@wildky are you sure it was TypeScript compiler itself that was crashing in your case, like in original issue and not prisma itslef, like reported in #9372 ? I think they are 2 different issues, latter being successfully reproduced. I could try with 250k rows, sure, but i am not quite sure it is reasonable to expect TS compiler to be able to handle 250k compile-time errors, even 1000 is kind of a stretch.

Also, note, that for example, this code, will still produce only a single compile-time error, despite inserting 250000 rows:

```ts
prisma.model.createMany(Array.from({length: 250_000}).map(() => ({
    intField: 'notAnIntValue'
})))
```

**Comment by wildky** [comment]
@SevInf Gotcha -- the latest comment (from just a few hours ago) in #9372 seems to infer it is not the prisma engine, but tsc compiler, so maybe they are the same issue after all 😆

**Comment by janpio** [comment]
That comment actually meant Typescript in Prisma Client - and the issue now closed as @SevInf could fix it. @GitMarijn @wildky Could you please take another look? We would really like to be able to reproduce this. Otherwise we will unfortunately have to close the issue. Thanks.

**Comment by janpio** [comment]
Ping @GitMarijn @wildky - can you help us out please? Thanks.

**Comment by janpio** [comment]
Closing then until someone provides more information. Please comment or open a new issue. Thanks!
